### ISSUE TITLE
[build] undefined reference to `llvm::Attribute::getAsString(bool) const'

### ISSUE THREAD

**Issue by erickq** [AliveToolkit/alive2#936]
![image](https://github.com/AliveToolkit/alive2/assets/12108571/03ef9317-f336-4c31-9c19-2c7d9a572f86)

![image](https://github.com/AliveToolkit/alive2/assets/12108571/368f6d59-03d4-4f5d-8e69-ec9297367394)
Obviously I can find this definition in the include folder, why does the link report an error?

**Comment by erickq** [comment]
> You must have a mismatch between the headers and the libraries. Probably you did 'git pull' on the llvm dir but didn't build it again?

Pure community code, I didn't make any changes.@nunoplopes

**Comment by nunoplopes** [comment]
It's a linker error, so it's a problem on your end: you're mixing binaries and headers from different places. Check the cmake log and make sure you've compiled LLVM after the last 'git pull'.

**Comment by erickq** [comment]
> It's a linker error, so it's a problem on your end: you're mixing binaries and headers from different places. Check the cmake log and make sure you've compiled LLVM after the last 'git pull'.

Yes, I did. but it's weird

**Comment by erickq** [comment]
> It's a linker error, so it's a problem on your end: you're mixing binaries and headers from different places. Check the cmake log and make sure you've compiled LLVM after the last 'git pull'.

The mistakes still exist.

**Comment by erickq** [comment]
![image](https://github.com/AliveToolkit/alive2/assets/12108571/efcd06e2-972e-42b1-a119-19e5be9d9842)

Is LLVM_DIR correctly configured?@nunoplopes

**Comment by nunoplopes** [comment]
> ![image](https://user-images.githubusercontent.com/12108571/270094829-efcd06e2-972e-42b1-a119-19e5be9d9842.png)
> 
> Is LLVM_DIR correctly configured?@nunoplopes

Depends on where you've your LLVM. Mine is:
```
LLVM_DIR =/home/nuno/llvm/build/lib/cmake/llvm
```

**Comment by regehr** [comment]
I agree that this has to be an LLVM mismatch, I just built a fresh LLVM+Alive2 for our compiler explorer instance and would have seen this error, if it existed

**Comment by regehr** [comment]
@erickq is this a scratch machine where you can play around and start uninstalling llvms that are supposed to be unrelated here? ideally get rid of all of them except the fresh one
